pkg/services/scm_provider/utils.go

```diff
@@ -114,10 +155,18 @@ func ListRepos(ctx context.Context, provider SCMProviderService, filters []argop
 		}
 	}
 
-	repos, err = getBranches(ctx, provider, filteredRepos, compiledFilters)
+	pullRequestRepos, err := getPullRequests(ctx, provider, filteredRepos, compiledFilters)
 	if err != nil {
 		return nil, err
 	}
+
+	branchRepos, err := getBranches(ctx, provider, filteredRepos, compiledFilters)
```

Comment:
I'm growing progressively more uncomfortable with the "repos" struct being used for non-repo stuff like pull requests and branches. But that's probably a refactor for another day. :-)

Reply:
I completely agree. It did not feel 100% right doing this. How about renaming the `Repository` struct to `argoapp`, since that's basically what it ends up being?

Reply:
Yep, I like that idea. Maybe something like `SCMGeneratorApp` to differentiate from the eventual actual App? idk, at your discretion. :-)

Reply:
I'll do this in a follow-up PR. 👍🏼